.github/workflows/ci.yaml

```diff
@@ -52,13 +52,50 @@ jobs:
       - name: Test with Coverage
         run: poetry run pytest --cov=. --cov-report=xml
         if: matrix.python-version == '3.10'
+      - name: Test Examples
+        run: poetry run pytest examples/
+        if: matrix.python-version == '3.10'
       - uses: actions/upload-artifact@v3
         with:
           name: coverage-xml
           path: coverage.xml
         if: matrix.python-version == '3.10'
+  test_minimal:
+    # tests the hello-world example against a minimal install
+    needs: validate
+    runs-on: ubuntu-latest
+    env:
+      python_version: "3.10"
```

Comment:
why do you need a different job for this? cant you testit in the `Test Examples` stage above?

Reply:
It is because the Test Examples job runs on environment with full extras: `run: poetry install --no-interaction --no-root --extras="full"`.

This is to explicitly test the app on the minimal install, so different environment required.

Reply:
The dev dependencies are equal to full extras.

Reply:
Well all except Brotli, but I'll admit that once I saw brotli wasn't there I assumed that the other optionals weren't there either. I'll refactor this. So just oversight that brotli isn't there?

Reply:
Yup